benchmarks/domains/Hardness.py

```diff
@@ -0,0 +1,266 @@
+# Hardness benchmarking, a maximization task on experimental hardness dataset.

```

Comment:
You have a longer explanation in the `__main__` section of the script. That should go here, and you can then print it from main via the `__doc__` attribute of the file. Overall, it should become very clear from the text what is the goal of this benchmark. Details that are not relevant to understand the overall task, e.g. how exactly the data is loaded (for example, that you consider only contexts with more then 5 points etc) should not be mentioned here but in their respective code section. For example, for the data loading, you'd need to add some data loading function whose docstring/comments explain it.